Sources/_RegexParser/Regex/Parse/Parse.swift

```diff
@@ -583,5 +583,15 @@ public func parseWithDelimiters<S: StringProtocol>(
   _ regex: S
 ) throws -> AST where S.SubSequence == Substring {
   let (contents, delim) = droppingRegexDelimiters(String(regex))
-  return try parse(contents, defaultSyntaxOptions(delim, contents: contents))
+  do {
+    return try parse(contents, defaultSyntaxOptions(delim, contents: contents))
+  } catch let error as Source.LocatedError<ParseError> {
+    // Convert the range in 'contents' to the range in 'regex'.
+    let delimCount = delim.opening.count
+    let offsets = contents.offsets(of: error.location.range)
+    let startIndex = regex.index(atOffset: delimCount + offsets.lowerBound)
+    let endIndex = regex.index(atOffset: delimCount + offsets.upperBound)
+
+    throw error.error.addingLocation(startIndex..<endIndex)
+  }
```

Comment:
Would it be worth generalizing this to any `LocatedErrorProtocol`? Just to make sure if a different kind of error is thrown, we don't silently use the wrong location? Doesn't have to be done in this PR tho.

Reply:
> Just to make sure if a different kind of error is thrown, we don't silently use the wrong location? 

Good catch! Although we currently don't use other underlying errors than `ParserError`, to make it safe, I generalized it in this PR.